baserun/baserun.py

```diff
@@ -417,11 +421,9 @@ def trace(
 
     @staticmethod
     @contextmanager
-    def start_trace(
-        *args, name: Optional[str] = None, end_on_exit=True, **kwargs
-    ) -> Generator[Optional[Run], None, None]:
+    def start_trace(name: Optional[str] = None, end_on_exit=True, **kwargs) -> Generator[Run, None, None]:
         if not Baserun._initialized:
-            yield None
+            raise NotInitializedException
```

Comment:
We actually need to keep the `yield None` because we don't want to raise an error if Baserun is not initialized. We have many users who want to do tracing in dev/staging but not in production.

Reply:
I can't say I agree with that. You're not able to just not call in your dev/staging environment `Baserun.init()` and call it good. If you do that even behaviour you recomend on the docs would cause an error:
```py
def ask_question(question="What is the capital of the US?") -> str:
    with baserun.start_trace() as trace:
        # Your code here, for example:
        answer_id = answer_question(question)

        trace.result = my_result
```
if a trace is None then you'd get `AttributeError: 'NoneType' object has no attribute 'result'`

summing up:
would it be nice to be able to "switch off" baserun for dev env? - definitely
is this a way to go to achieve it - absolutely not

Reply:
That's the issue that should be fixed, then. Instead of returning `None` we need to return some sort of object that can be assigned to in all of the ways that are supported in our documentation.

Reply:
That is-- today the only thing that's broken is assigning to `trace`, whereas when raising an error everything will be broken

Reply:
> the only thing that's broken is assigning to `trace`
nope, not true. there are many side effects of not initializing baserun right now.

I'm closing this because in https://github.com/baserun-ai/baserun-py/pull/129 I took things here a bit further so it's easier to make it work there